### PR TITLE
Add validation to encrypted custom fields where needed

### PR DESCRIPTION
**Pull request by inietov**
# Description
A clone of #12681 with the new importer. Don't know how get the validation errors to display correctly :(

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by marcusmoore** [comment]
@inietov is the issue with validation issues displaying in the importer or on the create and update pages?

**Comment by inietov** [comment]
Thanks for lend me another pair of eyes.
In the importer, I'm able to display that an error occurred, but then when I'm trying to show more details like what field and value triggered the error, I don't know how to get them to display there.

**Comment by marcusmoore** [comment]
I was testing out how importer errors are displayed and noticed that I'm not able to get errors out of the importer for _non-encrypted_ custom fields (even on `develop`). Going with the default seeded Mac Address custom field, I can include `00` for that field in a csv and it enters the database just fine. Do you see the same thing?

**Comment by spencerrlongg** [comment]
I hadn't completed my testing yet, but I had the same experience, I was going to double check when I had time to make sure before I commented.

File: app/Importer/AssetImporter.php [hunk 1]
```diff
@@ -21,11 +22,24 @@ protected function handle($row)
         parent::handle($row);
 
         if ($this->customFields) {
+            $input = array();
```
Comment (inietov) [inline app/Importer/AssetImporter.php]: @marcusmoore @spencerrlongg  Sorry to invoke you, just want to leave a reminder here to see on monday if needed. We set the CustomFields here, but they aren't tied to an asset, so I had to manually retrieve the model to access the validation rules and validate the fields before we create the asset to save it. So I create a custom validator that supposedly shows the error and then stop the import. But I can only 'skip' silently the rows that doesn't meet the criteria, without an actual error.

**Comment by inietov** [comment]
So, I notice that even in not encrypted custom fields, the errors are not always reported back, so I thought that maybe the importer needs it's own PR, but in the meantime it was slowing down the other part that closes a lot of my assigned issues.

**Review by marcusmoore — APPROVED** [review]
This was tricky @inietov. Good work finding a solution. Unsetting rules makes me go 😬 but I understand why it works in this case.

I tested this and it works but we have to keep in mind that we don't have a fix in for the importer yet so merging this before addressing that could cause some issues down the line since we mess with the Asset model rules in this PR.

**Comment by inietov** [comment]
> This was tricky @inietov. Good work finding a solution. Unsetting rules makes me go 😬 but I understand why it works in this case.
> 
> I tested this and it works but we have to keep in mind that we don't have a fix in for the importer yet so merging this before addressing that could cause some issues down the line since we mess with the Asset model rules in this PR.

Yeah, right. That was why I started working in the importer validation. Ugh, sorry. Let me think about this again, from the top.

**Comment by snipe** [comment]
@inietov Should we close this? I haven't seen any updates in a bit.

**Comment by inietov** [comment]
Yeah, that's fair. I think I need to rethink about these use cases.

**Pull request closed without merging**